x-pack/plugins/data_visualizer/public/application/data_drift/document_count_chart_single_brush/single_brush.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef, type FC } from 'react';
+
+import * as d3Brush from 'd3-brush';
+import * as d3Scale from 'd3-scale';
+import * as d3Selection from 'd3-selection';
+import * as d3Transition from 'd3-transition';
+import './single_brush.scss';
+
+const { brush, brushSelection, brushX } = d3Brush;
+const { scaleLinear } = d3Scale;
+const { select: d3Select } = d3Selection;
+// Import fix to apply correct types for the use of d3.select(this).transition()
+d3Select.prototype.transition = d3Transition.transition;
+
+export const getSingleBrushWindowParameters = (
+  clickTime: number,
+  minTime: number,
+  maxTime: number
+): SingleBrushWindowParameters => {
+  const totalWindow = maxTime - minTime;
+
+  // min deviation window
+  const minDeviationWindow = 10 * 60 * 1000; // 10min
+  const minBaselineWindow = 30 * 60 * 1000; // 30min
+  const minWindowGap = 5 * 60 * 1000; // 5min
+
+  // work out bounds as done in the original notebooks,
+  // with the deviation window aiming to be a 1/10
+  // of the size of the total window and the baseline window
+  // being 3.5/10 of the total window.
+  const deviationWindow = Math.max(totalWindow / 10, minDeviationWindow);
+  const baselineWindow = Math.max(totalWindow / 3.5, minBaselineWindow);
+  const windowGap = Math.max(totalWindow / 10, minWindowGap);
+
+  const deviationMin = clickTime - deviationWindow / 2;
+
+  const baselineMax = deviationMin - windowGap;
+  const baselineMin = baselineMax - baselineWindow;
+
+  return {
+    min: Math.round(baselineMin),
+    max: Math.round(baselineMax),
```

Comment:
Suggest to use the deviation time range here. The baseline is a time range before the deviation and has quite an offset to where the user clicked. See the following example, I'm clicking on the spike in the dataset but the selected time range will be way behind it:

![ml-data-drift-0001](https://github.com/elastic/kibana/assets/230104/66c4a2c9-e2af-4dec-b8da-451a11a54cd0)